app/src/main/java/mollie/tictactoe/ui/UIBoardManager.java

```diff
@@ -0,0 +1,50 @@
+package mollie.tictactoe.ui;
+
+import android.content.Context;
+import android.widget.Toast;
+
+import ttt.game.Board;
+import ttt.game.Marks;
+
+public class UIBoardManager {
+
+    public static void updateUIBoard(Board board, BoardView view) {
+        for (int i = 0; i < board.size(); i++) {
+            CellButton button = (CellButton) view.findViewWithTag(String.valueOf(i));
```

Comment:
This could be pushed down into the board view:
`CellButton button = view.findById(i)`